compiler/rustc_middle/src/query/plumbing.rs

```diff
@@ -382,19 +433,19 @@ macro_rules! define_callbacks {
             #[inline(always)]
             pub fn $name(self, key: query_helper_param_ty!($($K)*)) -> $V
             {
-                restore::<$V>(query_get_at(
+                restore::<$V>(with_query_caches!(query_get_at(
```

Comment:
This change is probably a bit costly as is doubles the size of the query accessor.

Reply:
Refers to the rustc-perf's result, this change doesn't cause regression?

Reply:
That doesn't really affect the code size.